app/src/main/java/com/cornellappdev/coffee_chats_android/SchedulingActivity.kt

```diff
@@ -58,40 +61,70 @@ class SchedulingActivity :
                 setUpNetworking(preferencesHelper.accessToken!!)
                 try {
                     user = getUser()
-                } catch (e : Exception) {
-                    // login error, prompt user to sign in
-                    signIn()
-                }
-                // move to onboarding if user hasn't finished
-                if (!user.hasOnboarded) {
-                    val intent = Intent(applicationContext, OnboardingActivity::class.java)
-                    intent.putExtra(ACCESS_TOKEN_TAG, preferencesHelper.accessToken!!)
-                    startActivity(intent)
-                } else {
-                    setUpDrawerLayout()
-                    if (user.currentMatch == null) {
-                        primaryActionButton.visibility = View.GONE
-                        ft.add(fragmentContainer.id, NoMatchFragment()).addToBackStack(noMatchTag)
-                        ft.commit()
-                        headerText.text = getString(R.string.no_match_header)
+                    // move to onboarding if user hasn't finished
+                    if (!user.hasOnboarded) {
+                        val intent = Intent(applicationContext, OnboardingActivity::class.java)
+                        intent.putExtra(ACCESS_TOKEN_TAG, preferencesHelper.accessToken!!)
+                        startActivity(intent)
                     } else {
-                        ft.add(
-                            fragmentContainer.id,
-                            ProfileFragment.newInstance(user.currentMatch!!.matchedUser)
-                        ).addToBackStack(matchTag)
-                        ft.commit()
-                        headerText.text = getString(R.string.match_header)
+                        setUpDrawerLayout()
+                        val c = this@SchedulingActivity
+                        val isUserMatched = user.currentMatch != null
+                        if (!isUserMatched) {
+                            primaryActionButton.visibility = View.GONE
+                        }
+                        viewPager.adapter =
+                            ViewPagerAdapter(c, isUserMatched)
+                        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+                            tab.text =
+                                if (position == 0) c.getText(R.string.match_header)
+                                else c.getText(R.string.people_header)
+                        }.attach()
+                        // resize tabs so they wrap tab text
+                        tabLayout.apply {
```

Comment:
does the tabLayout need to be resized because it depends on some fields that are only initialized when the Activity is initialized?

Reply:
Ahh wait I think I understand

Reply:
Yeah, I added this so that the `TabLayout` wouldn't be split exactly in half, since the left tab's text is longer.